### ISSUE TITLE
About the wavegru architecture

### ISSUE THREAD

**Issue by loopzy** [google/lyra#41]
hi, I am little confused about the wavegru built in the code  and I can't get the meaning of the rows and cols of the ar_to_gate layer and gru layer? It seems that the ar_to_gate and conditioning stack both have output with dim=3*1024. but the in_channel of gru layer is 1024... Is the model architecture different from the paper "Generative Speech Coding with Predictive Variance Regularization."?

**Comment by Yablon** [comment]
@loopzy the number 3 comes from the calculation of RNN. You can see the following image about how torch.nn.LSTMCell is calculated.
![image](https://user-images.githubusercontent.com/38012559/119936864-6689dd00-bfbc-11eb-8ff4-17a52e0399a9.png)

where x is calculated by three matrix, Wii, Wif, Wig，which can fused together. 

That makes the number 3.

**Comment by loopzy** [comment]
> @loopzy the number 3 comes from the calculation of RNN. You can see the following image about how torch.nn.LSTMCell is calculated.
> ![image](https://user-images.githubusercontent.com/38012559/119936864-6689dd00-bfbc-11eb-8ff4-17a52e0399a9.png)
> 
> where x is calculated by three matrix, Wii, Wif, Wig，which can fused together.
> 
> That makes the number 3.

Thanks! I guessed that too. But in that way, the 3*1024 vector is a combination of 3 copies of 1024 vector and I am not sure.

Also, it is strange that he first conv1d of conditioning stack has kernel_size=3 while the model  accepts only one logmel feature in each time. Do you have any ideas about that?

**Comment by NTT123** [comment]
@loopzy, kernel_size = 3 means that the first conv1d layer looks at 3 logmel frames: 1 logmel frame in the future (a frame look-ahead), 1 frame in the present and 1 frame in the past.

RNN layer takes 1 feature vector each time but this feature vector can be computed from 1 logmel frame look ahead and many frames at present and in the past

**Comment by loopzy** [comment]
> @loopzy, kernel_size = 3 means that the first conv1d layer looks at 3 logmel frames: 1 logmel frame in the future (a frame look-ahead), 1 frame in the present and 1 frame in the past.
> 
> RNN layer takes 1 feature vector each time but this feature vector can be computed from 1 logmel frame look ahead and many frame

**Comment by loopzy** [comment]
> @loopzy, kernel_size = 3 means that the first conv1d layer looks at 3 logmel frames: 1 logmel frame in the future (a frame look-ahead), 1 frame in the present and 1 frame in the past.
> 
> RNN layer takes 1 feature vector each time but this feature vector can be computed from 1 logmel frame look ahead and many frames at present and in the past

Thanks a lot！
